src/pveclib/vec_int512_ppc.h

```diff
@@ -840,134 +625,100 @@ vec_mul128x128_PWR9 (vui128_t m1l, vui128_t m2l)
-CFLAGS-vec_runtime_PWR8.c += -mcpu=power8
-CFLAGS-vec_runtime_PWR9.c += -mcpu=power9
- * \endcode
- *
- * \todo Is there a way for automake to compile vec_int512_runtime.c
- * with -mcpu=power7 with -o vec_runtime_PWR7.o.
- * And similarly for PWR8/PWR9.
- *
- * \subsection i512_libary_issues_0_0_1 Calling Multi-platform functions
- *
- * The next step is to provide mechanisms for applications to call
- * these functions via static or dynamic linkage.
- * For static linkage the application needs to reference a specific
- * platform variant of the functions name.
- * For dynamic linkage we will use <B>STT_GNU_IFUNC</B> symbol resolution
- * (a symbol type extension to the ELF standard).
- *
- * \subsubsection i512_libary_issues_0_0_1_1 Static linkage to platform specific functions
+ * This ensures that target specific runtime
+ * implementations have unique function symbols. This is important to
+ * avoid linker errors (due to duplicate symbol names).
+ *
+ * \note Each runtime operation will have 2 or 3 target qualified
+ * implementations. This is times 2 with separate builds for static
+ * archives and dynamic (DSO) libraries.
+ * The big endian powerpc64 platform supports 3
+ * VSX enabled targets -mcpu=[power7|power8|power9].
+ * The little endian powerpc64le platform currently supports 2
+ * VSX enabled targets -mcpu=[power8|power9]. POWER7 is not
+ * supported for powerpc64le and the vec_runtime_PWR7.c source
+ * files are conditionally nulled out for powerpc64le targets.
+ *
+ * \subsection i512_libary_issues_0_0_1_1 Static linkage to platform specific functions
  * For static linkage the application is compiled for a specific
  * platform target (via -mcpu=). So function calls should be bound to
  * the matching platform specific implementations. The application
  * may select the platform specific function directly by defining
  * a <I>extern</I> and invoking the platform qualified function.
  *
- * Or simply use the __VEC_PWR_IMP() macro as wrapper for the function
- * name in the application.
- * This selects the appropriate platform specific implementation based
- * on the -mcpu= specified for the application compile.
- * For example.
+ * For applications binding to PVECLIB via static archives
+ * it is convenient to apply the __VEC_PWR_IMP() wrapper to the
+ * function call:
  * \code
   k = __VEC_PWR_IMP (vec_mul128x128)(i, j);
  * \endcode
+ * The function call symbol picks up the target suffix based on
+ * the compile target (\-mcpu=) for the application
+ * (see \ref main_libary_issues_0_0_1_1).
+ * The linker will extract the matching implementations from the
+ * PVECLIB archive and (statically) bind them with the application.
+ * This simplifies binding the application to the matching target
+ * specific implementations.
+ *
+ * \subsection i512_libary_issues_0_0_1_2 Dynamic linkage to platform specific functions
+ *
+ * For applications binding to dynamic libraries, the target qualified
+ * naming strategy,
```

Comment:
delete that trailing comma. optionally rewrap.

Reply:
OK